VersionControlUI/src/DiffManager.cpp

```diff
@@ -0,0 +1,300 @@
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#include "DiffManager.h"
+
+#include "VisualizationBase/src/VisualizationManager.h"
+#include "VisualizationBase/src/items/ViewItem.h"
+#include "VisualizationBase/src/nodes/ViewItemNode.h"
+#include "VisualizationBase/src/ViewItemManager.h"
+#include "VisualizationBase/src/declarative/GridLayouter.h"
+#include "VisualizationBase/src/overlays/HighlightOverlay.h"
+
+#include "FilePersistence/src/version_control/GitRepository.h"
+#include "FilePersistence/src/simple/SimpleTextFileStore.h"
+
+#include "OOModel/src/declarations/Project.h"
+#include "OOModel/src/statements/ExpressionStatement.h"
+#include "OOModel/src/expressions/EmptyExpression.h"
+
+#include "ModelBase/src/model/AllTreeManagers.h"
+#include "ModelBase/src/model/TreeManager.h"
+#include "ModelBase/src/nodes/Reference.h"
+
+
+namespace VersionControlUI
+{
+
+DiffManager::DiffManager(QString oldVersion, QString newVersion, QString project)
+{
+	_oldVersion = oldVersion;
+	_newVersion = newVersion;
+	_project = project;
+}
+
+void DiffManager::visualize()
+{
+	auto diffViewItem = Visualization::VisualizationManager::instance().mainScene()->
+			viewItems()->newViewItem("DiffView");
+
+	QString projectsDir = "projects/" + _project;
+
+	// get GitRepository
+	if (!FilePersistence::GitRepository::repositoryExists(projectsDir))
+	{
+		qDebug() << "no repository found at " + projectsDir;
+		return;
+	}
+
+	FilePersistence::GitRepository repository{projectsDir};
+
+	// load name into tree
+	std::unique_ptr<const FilePersistence::Commit> oldCommit{repository.getCommit(_oldVersion)};
```

Comment:
All this commit, store and manager stuff is repetitive, can't we abstract it to reduce duplication?